### PR TITLE
Use test-ci k8s cluster; add Istio integration test

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: tests/integration/istio/istio_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		t.Skipf("Skipping test due to missing KUBERNETES_CONTEXT variable")
+	}
+
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir:          "step1",
+		Dependencies: []string{"@pulumi/kubernetes"},
+		Quick:        true,
+		SkipRefresh:  true,
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			// assert.NotNil(t, stackInfo.Deployment)
```
Comment (lblackstone) [inline tests/integration/istio/istio_test.go]: Is this still needed?

File: tests/integration/istio/istio_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			// assert.Equal(t, 143, len(stackInfo.Deployment.Resources))
+
+			frontend := stackInfo.Outputs["frontendIp"].(string)
+			for range []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10} {
```
Comment (lblackstone) [inline tests/integration/istio/istio_test.go]: Any reason not to use a simple for loop?

```suggestion
			for i := 1; i <= 10; i++ {
```

File: tests/integration/istio/step1/index.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    { provider: k8sProvider }
+);
+
+function addNamespace(o: any) {
+    if (o !== undefined) {
+        o.metadata.namespace = "bookinfo";
+    }
+}
+
+const bookinfo = new k8s.yaml.ConfigGroup(
```
Comment (lblackstone) [inline tests/integration/istio/step1/index.ts]: Is `ConfigGroup` rather than `ConfigFile` needed because of istio?

Comment (hausdorff) [reply tests/integration/istio/step1/index.ts]: I forgot about this, this is from when we were reading the file manually and using `istioctl` to inject the sidecar.

**Comment by hausdorff** [comment]
@lblackstone ok we're probably ready to go once CI passes.

**Pull request merged**